web_client/src/services/api.js

```diff
@@ -0,0 +1,94 @@
+import request from 'superagent';
+
+const CLIENT_ID = 'b231329aba1a4c539375436a267db917';
+//const REDIRECT_URI = 'http://127.0.0.1:3000';
```

Comment:
can' you just use `/` and it'll know the hostname of the server? you can also pass down the host name in meta data and store it as environmental variable somewhere

Reply:
so i need the full path because of the way the spotify authorization flow works
1- i redirect the user to the spotify authorization website, passing the redirect uri over as a query param, so once the user logs in the spotify page sends them back to playlistsouffle.com (in this case 127.0.0.1:3000)
2- (i have to manually add a list of whitelisted redirect uris on my spotify developers dashboard page)
3- when i then request a refreshtoken from the playlist souffle `/refreshtoken` endpoint (using my server-side spotify client secret), i need to also send the same redirect uri that i used for the authorization page redirect, or else spotify will reject the request.

Reply:
got it. yeah, i think you need to generate the hostname on the server.

I'm not sure. (and am in a bit of a rush right now.) is this useful?

https://github.com/react-boilerplate/react-boilerplate/issues/1250

(let me know what you come up with!)